### PR TITLE
Add DisableStoreOutput for Responses agents

### PR DESCRIPTION
**Pull request by qmuntal**
## Summary
This ports the .NET SDK's stored-output-disabled Responses behavior to Go, so apps that manage their own local history can opt out of service-side Responses output storage in the same way.

The change adds `DisableStoreOutput` to OpenAI and Foundry Responses-backed agent configs, sends `store=false` when enabled, includes `reasoning.encrypted_content` for stateless reasoning continuity, and avoids saving response IDs into sessions that cannot safely continue from non-stored outputs. Per-run `ResponsesNewParams.Store` is also supported and takes precedence over the instance-level `DisableStoreOutput` setting for that run.

It also teaches the Responses provider to replay local assistant messages and function-call history in a Responses-compatible shape while preserving the original assistant content order, then updates the related examples to use local history with output storage disabled.

## Validation
- `go test -count=1 ./provider/openaiprovider ./provider/foundryprovider ./message/messagefilter ./examples/02-agents/agents/step07_3rdparty_session_storage ./examples/02-agents/agents/step17_additional_ai_context ./examples/02-agents/providers/azure/openai_responses`

### CODE REVIEW

File: provider/openaiprovider/responses.go [hunk 1]
```diff
@@ -506,12 +512,15 @@ func responsesBuildMessageParam(msg *message.Message, resp responses.ResponseInp
 		}
 
 	case message.RoleAssistant:
+		var outputContents []responses.ResponseOutputMessageContentUnionParam
 		for _, c := range msg.Contents {
 			switch c := c.(type) {
 			case *message.TextContent:
```
[the remark this answers is not in the document]
Comment (qmuntal) [reply provider/openaiprovider/responses.go:544]: Fixed in the latest push. Assistant replay now flushes accumulated output-text content before appending function-call or reasoning input items, so the original message content order is preserved. I also added a regression test for text -> function_call -> text ordering.

**Review by gdams — APPROVED** [review]

File: provider/openaiprovider/responses.go [hunk 2]
```diff
@@ -216,12 +220,15 @@ func (a *responsesClient) run(ctx context.Context, messages []*message.Message,
...
 // buildCompletionParams constructs the parameters for the OpenAI chat completion API.
-func responsesBuildCompletionParams(model string, messages []*message.Message, opts []agent.Option) (responses.ResponseNewParams, error) {
+func responsesBuildCompletionParams(config AgentConfig, messages []*message.Message, opts []agent.Option) (responses.ResponseNewParams, error) {
 	var params responses.ResponseNewParams
 	if p, ok := agent.GetOption(opts, ResponsesNewParams); ok {
 		params = p
 	}
-	params.Model = cmp.Or(params.Model, model)
+	if config.DisableStoreOutput && param.IsOmitted(params.Store) {
+		params.Store = openai.Bool(false)
```
[the remark this answers is not in the document]
Comment (qmuntal) [reply provider/openaiprovider/responses.go]: Fixed in the latest push. When the effective request has store=false (from either AgentConfig.DisableStoreOutput or per-run ResponsesNewParams.Store=false), the provider now includes reasoning.encrypted_content unless it is already present. I also added tests for config-level store=false, per-run store=false, and duplicate prevention.
